dellyGermline.wdl

```diff
@@ -0,0 +1,262 @@
+version 1.0
+
+struct inputSamples {
+  File bam
+  File bai
+  File vcf
+}
+
+struct dellyResources {
+    String modules
+    String referenceGenome
+    String dellyExclude
+}
+
+workflow dellyGermline {
+  input {
+    String reference
+    Array[inputSamples] inputSamples
+  }
+
+  parameter_meta {
+    reference: "The genome reference build. for example: hg19, hg38"
+    inputSamples: "Collection of BAM, BAI, and VCF files for >= 20 samples"
+  }
+
+  Map[String,dellyResources] resources = {
+    "hg19": {
+      "modules": "delly/0.9.1 bcftools/1.9 tabix/0.2.6 hg19/p13 hg19-delly/1.0",
+      "referenceGenome": "$HG19_ROOT/hg19_random.fa",
+      "dellyExclude": "$HG19_DELLY_ROOT/human.hg19.excl.tsv"
+    },
+    "hg38": {
+      "modules": "delly/0.9.1 bcftools/1.9 tabix/0.2.6 hg38/p12 hg38-delly/1.0",
+      "referenceGenome": "$HG38_ROOT/hg38_random.fa",
+      "dellyExclude": "$HG38_DELLY_ROOT/human.hg38.excl.tsv"
+    }
+  }   
+
+
+  scatter(s in inputSamples) {
```

Comment:
This scatter only creates three strings? Does not call any task?

Reply:
Yep! This is because there are three input files for each sample, which come in in as an array. Then I use scatter to split the files into their respective types (bam, bai, and vcf), ensuring that the sample order for each list is the same. Then I can use these lists for tasks been called below.

Reply:
You can see Peter's mavis workflow for another example of this: https://github.com/oicr-gsi/mavis/blob/master/mavis.wdl#L43

Reply:
It's OK, seems wdl has no other syntax to create array of string without scatter.